### DISCUSSION TITLE
How to use specifically node version of axios in ts?

### DISCUSSION THREAD

**Discussion by Haiz14** [axios/axios#6594] [Q&A]
Wrote a simple axios script (required import was used) with authorisation header, works fine. When the same script is being used in  [twilio-node](https://github.com/twilio/twilio-node) (ts version) and it is complained that Authorization header is not present.

Probably a cors issue. Since twilio-node runs on node, why can't axios use node version of axios instead of browser. 

How to specify to use node version of axios in type script? Example in required mode (doesn't work on ts) 

`require('axios/dist/node/axios.cjs')`

**Comment by truonganDevOps** [comment]
In Axios v1.x, you can explicitly force the **Node.js HTTP adapter** directly in your config — no need to import from dist paths:

```typescript
import axios from "axios";

const instance = axios.create({
  adapter: "http", // force Node.js HTTP adapter
  headers: {
    Authorization: `Bearer ${yourToken}`,
  },
});
```

**Why your workaround does not work in TypeScript:**

`require("axios/dist/node/axios.cjs")` is a CommonJS path that works in `.js` but fails in strict TypeScript because there are no type declarations for that subpath.

**Root cause of your CORS/Authorization issue:**

When running inside `twilio-node` (which is a Node.js environment), Axios should automatically pick the `http` adapter. If the `Authorization` header is still missing, the likely culprit is that somewhere in the chain a CORS preflight (`OPTIONS`) request is being triggered — browsers strip custom headers from preflight requests, but in pure Node.js there is no browser CORS at all.

Check if:
1. The `Authorization` header is being removed by a request interceptor in `twilio-node`
2. You are accidentally running this in a browser bundler context (Webpack/Vite) where the browser adapter is selected instead

Setting `adapter: "http"` explicitly guarantees Node.js behavior regardless of the environment detection.
